catroid/src/org/catrobat/catroid/content/bricks/StopScriptBrick.java

```diff
@@ -0,0 +1,179 @@
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.content.bricks;
+
+import android.content.Context;
+import android.content.res.ColorStateList;
+import android.graphics.drawable.Drawable;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.BaseAdapter;
+import android.widget.CompoundButton;
+import android.widget.CompoundButton.OnCheckedChangeListener;
+import android.widget.Spinner;
+import android.widget.TextView;
+
+import com.badlogic.gdx.scenes.scene2d.actions.SequenceAction;
+
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.common.BrickValues;
+import org.catrobat.catroid.common.SoundInfo;
+import org.catrobat.catroid.content.Sprite;
+
+import java.util.List;
+
+public class StopScriptBrick extends BrickBaseType {
+	private static final long serialVersionUID = 1L;
+
+	private String[] spinnerValue;
+	private int spinnerSelectionId;
+
+	private SoundInfo sound;
+	private transient SoundInfo oldSound;
+	private transient AdapterView<?> adapterView;
+
+	public StopScriptBrick(int spinnerSelection) {
+		spinnerValue = new String[3];
+		spinnerSelectionId = spinnerSelection;
+	}
+
+//	@Override
+//	public int getRequiredResources() {
+//		return NO_RESOURCES;
+//	}
+
+	@Override
+	public View getViewWithAlpha(int alphaValue) {
+		if (view != null) {
+			View layout = view.findViewById(R.id.brick_stop_script_layout);
+			Drawable background = layout.getBackground();
+			background.setAlpha(alphaValue);
+
+			Spinner spinner = (Spinner) view.findViewById(R.id.brick_stop_script_spinner);
+			spinner.getBackground().setAlpha(alphaValue);
+
+			TextView textStopScript = (TextView) view.findViewById(R.id.brick_stop_script_label);
+//			ColorStateList color = textStopScript.getTextColors().withAlpha(alphaValue);
+			textStopScript.setTextColor(textStopScript.getTextColors().withAlpha(alphaValue));
+
+//			if(adapterView != null) {
+//				((TextView) adapterView.getChildAt(0)).setTextColor(color);
+//			}
+
+			this.alphaValue = alphaValue;
+		}
+		return view;
+	}
+
+	@Override
+	public View getView(Context context, int brickId, BaseAdapter baseAdapter) {
+		if (animationState) {
+			return view;
+		}
+		view = View.inflate(context, R.layout.brick_stop_script, null);
+		view = getViewWithAlpha(alphaValue);
+		setCheckboxView(R.id.brick_stop_script_checkbox);
+
+		final Brick brickInstance = this;
+		checkbox.setOnCheckedChangeListener(new OnCheckedChangeListener() {
+			@Override
+			public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
+				checked = isChecked;
+				adapter.handleCheck(brickInstance, isChecked);
+			}
+		});
+
+		Spinner spinner = (Spinner) view.findViewById(R.id.brick_stop_script_spinner);
+		if(!(checkbox.getVisibility() == view.VISIBLE)) {
```

Comment:
This would be more readable:

```java
if(checkbox.getVisibility() == view.VISIBLE) {
  spinner.setClickable(false);
  spinner.setEnabled(false);
} else {
  spinner.setClickable(true);
  spinner.setEnabled(true);
}
```

or this would work too:

```java
bool isVisible = checkbox.getVisibility() == view.VISIBLE;
spinner.setClickable(isVisible);
spinner.setEnabled(isVisible);
```